rotkehlchen/chain/evm/decoding/curve/decoder.py

```diff
@@ -602,13 +701,29 @@ def _decode_gauge_events(self, context: DecoderContext) -> DecodingOutput:
         for event in context.decoded_events:
             if (
                 event.location_label == provider and
-                event.address == gauge_address and
+                (event.address == gauge_address or event.address in self.curve_deposit_contracts) and  # noqa: E501
                 event.asset.is_evm_token() and
-                event.balance.amount == asset_normalized_value(
-                    amount=raw_amount,
-                    asset=(evm_asset := event.asset.resolve_to_evm_token()),
+                (
+                    (
```

Comment:
you need comments here. This is a really complicated if. Can't it be written bit better? Least you can do is use this ample whitespace you are leaving here for some strategically places explainer comments.